test/browser/custom-elements.js

```diff
@@ -84,6 +84,93 @@ describe('Custom Elements', function () {
     domUI.dispose();
   });
 
+  it('should dispose vtree$ after the custom element has been destroyed', function () {
+    let log = [];
+    let number$ = Rx.Observable.range(1, 2).controlled();
+    let customElementSwitch$ = Rx.Observable.range(0, 2).controlled();
+    // Make simple custom element
+    Cycle.registerCustomElement('myelement', function () {
+      return {
+        vtree$: number$
+          .do(i => log.push(i))
+          .map(i => h('h3.myelementclass', String(i)))
+      };
+    });
+    // Use the custom element
+    let vtree$ = customElementSwitch$.map(theSwitch => {
+      return theSwitch === 0
+        ? h('div.toplevel', [h('myelement', {key: 1})])
+        : h('div.toplevel', []);
+    })
+    let domUI = Cycle.applyToDOM(createRenderTarget(), () => vtree$);
+    // Make assertions
+    customElementSwitch$.request(1);
+    number$.request(1);
+    let myElement = document.querySelector('.myelementclass');
+    assert.notStrictEqual(myElement, null);
+    assert.notStrictEqual(typeof myElement, 'undefined');
+    assert.strictEqual(myElement.tagName, 'H3');
+    assert.strictEqual(log.length, 1);
+
+    // Destroy the element
+    customElementSwitch$.request(1);
+    // Try to trigger onNext of myelement's vtree$
+    number$.request(1);
+    let destroyedElement = document.querySelector('.myelementclass');
+    assert.strictEqual(destroyedElement, null);
+    assert.notStrictEqual(log.length, 2);
+    domUI.dispose();
+  });
```

Comment:
It's actually a great idea to use controlled Observables for testing. Nice. I think I'll do this more in other tests too.